CSharp/DevVmPowershell/Helpers/Implementations/ProcessingHelper.cs

```diff
@@ -782,117 +427,385 @@ public async Task<bool> RemoveWorkerManagerServerFromDefaultResourcePoolAsync()
+			}
+			else
+			{
+				return -1;
+			}
+		}
 
-			// Setup for checking if Worker Manager Server exists
-			Relativity.Services.TextCondition conditionServer = new Relativity.Services.TextCondition()
+		private async Task CallCreateWorkerManagerServerAsync(HttpClient httpClient)
+		{
+			var workerManagerServerCreate = new
 			{
-				Field = Constants.Processing.NameField,
-				Operator = Relativity.Services.TextConditionEnum.EqualTo,
-				Value = Constants.Processing.ResourceServerName
+				resourceServer = new
+				{
+					Name = Constants.Processing.ResourceServerName,
+					IsDefault = true,
+					InventoryPriority = 100,
+					DiscoveryPriority = 100,
+					PublishPriority = 100,
+					ImageOnTheFlyPriority = 1,
+					MassImagingPriority = 100,
+					SingleSaveAsPDFPriority = 100,
+					MassSaveAsPDFPriority = 100,
+					URL = Constants.Processing.ResourceServerUrl
+				}
 			};
+			string request = JsonConvert.SerializeObject(workerManagerServerCreate);
+			HttpResponseMessage response = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourceServer.WorkerManagerCreateEndpointUrl, request);
+			if (!response.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to Create Worker Manager Server");
+			}
+		}
 
-			Relativity.Services.Query queryServer = new Relativity.Services.Query()
+		private async Task CallDeleteWorkerManagerServerAsync(HttpClient httpClient, int workerManagerServerArtifactId)
+		{
+			var workerManagerServerDelete = new
 			{
-				Condition = conditionServer.ToQueryString()
+				resourceServerID = workerManagerServerArtifactId
 			};
+			string request = JsonConvert.SerializeObject(workerManagerServerDelete);
+			HttpResponseMessage response = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourceServer.WorkerManagerDeleteEndpointUrl, request);
+			if (!response.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to Delete Worker Manager Server");
+			}
+		}
 
-			// Setup for checking if Worker Server exists
-			Relativity.Services.TextCondition conditionWorker = new Relativity.Services.TextCondition()
+		private async Task<int> GetWorkerServerArtifactIdAsync(HttpClient httpClient)
+		{
+			int workerServerArtifactId = -1;
+			var resourceServerQueryPayload = new
 			{
-				Field = Constants.Processing.NameField,
-				Operator = Relativity.Services.TextConditionEnum.EqualTo,
-				Value = Constants.Processing.ResourceServerName
+				Query = new
+				{
+					Condition = ""
+				}
 			};
+			string serverQuery = JsonConvert.SerializeObject(resourceServerQueryPayload);
+			HttpResponseMessage queryServerResponse = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.ResourceServer.QueryEndpointUrl, serverQuery);
+			if (!queryServerResponse.IsSuccessStatusCode)
+			{
+				throw new System.Exception("Failed to query for Worker Resource Server");

```

Comment:
Is there additional details in the HttpResponse regarding the actual error. We should probably put that in the Exception we are throwing which will be helpful when troubleshooting.

Reply:
Added failure message